core/src/main/java/kafka/server/SharePartitionManager.java

```diff
@@ -152,8 +163,13 @@ private void maybeProcessFetchQueue() {
                         topicIdPartition
                 );
                 SharePartition sharePartition = partitionCacheMap.computeIfAbsent(sharePartitionKey,
-                    k -> new SharePartition(shareFetchPartitionData.groupId, topicIdPartition, maxInFlightMessages, maxDeliveryCount,
-                            recordLockDurationMs, timer, time, persister, replicaManager));
+                    k -> {
+                        long start = time.milliseconds();
```

Comment:
Following on from @apoorvmittal10's point below, if we are going to use `time.hiResClockMs()`, we probably should be consistent to use it here also.